### PR TITLE
Add support for inverting and negating private keys

### LINKED ISSUE #124: Add support for calculating inverse of a private key
This can be useful for decrypting points multiplied by a key.

### PR DESCRIPTION
**Pull request by kripod**

### CODE REVIEW

**Comment by jprichardson** [comment]
FYI, @fanatid is AFK for 6 more days.

**Comment by fanatid** [comment]
Looks good on first view, thank you @kripod 
The main question, are we need this? Or this `can be useful for` (from issue #124)? Originally this library is bindings for [bitcoin-core/secp256k1](https://github.com/bitcoin-core/secp256k1) and that library is not export new functions.

Will review when return back, sorry for delay @kripod

**Comment by kripod** [comment]
Actually, I would need this functionality to implement a variant of mental poker.

File: lib/js/index.js [hunk 1]
```diff
@@ -19,6 +19,29 @@ exports.privateKeyExport = function (privateKey, compressed) {
...
+
+  if (bn.isZero()) {
+    return Buffer.alloc(32)
+  }
+
+  if (bn.ucmp(BN.n) !== 0) {
+    while (bn.isOverflow()) {
+      bn.isub(BN.n)
+    }
+  }
```
Comment (fanatid) [inline lib/js/index.js:33]: Why we need `while` loop here? Loop body will be executed 1 time at max..
```js
exports.privateKeyNegate = function (privateKey) {
  var bn = BN.fromBuffer(privateKey)
  if (bn.isZero()) return Buffer.alloc(32)

  if (bn.ucmp(BN.n) > 0) bn.isub(BN.n)
  return BN.n.sub(bn).toBuffer()
}
```

Comment (kripod) [reply lib/js/index.js:33]: You are right, the `while` statement can be changed to an `if` statement, and the condition can be concatenated with the previous `if`.

**Comment by kripod** [comment]
@fanatid Thank you for merging, I would highly appreciate if you could publish a new minor release soon.

File: lib/js/index.js (hunk 1 above)
Comment (fanatid) [reply lib/js/index.js:33]: Changed in https://github.com/cryptocoinjs/secp256k1-node/commit/b963603f2c0f556f572e6d1ac72cb054e5fce78d

**Comment by fanatid** [comment]
@kripod thank you for PR, will publish new version after CI tests for latest commit :)

**Pull request merged**